ghmirror/core/mirror_requests.py

```diff
@@ -58,6 +58,64 @@ def _cache_response(resp, cache, cache_key):
         cache[cache_key] = resp
 
 
+def _online_request(method, url, data=None,
+                    headers=None, parameters=None):
+    """
+    Handle API errors on conditional requests and try
+    to serve contents from cache
+    """
+    try:
+        resp = requests.request(method=method,
+                                url=url,
+                                headers=headers,
+                                data=data,
+                                timeout=REQUESTS_TIMEOUT,
+                                params=parameters)
+        resp.raise_for_status()
```

Comment:
I'm not convinced we should try to serve from cache on every 4xx. Most likely only in 403 and rate limited. But maybe you have already discussed this...

Reply:
That's what the code is doing, `raise_for_status()` raises an HTTPError exception if the status code of the request is 4xx or 5xx that we are capturing and trying to serve the content from the cache

Reply:
the code is catching every 4xx whereas I doubt we should serve from cache a request that's returning a 404, for example.